From a059a3fc9e94839c1df15b1b28cfa10e07dac084 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 1 Aug 2019 16:37:25 +0800
Subject: [PATCH 43/43] dix: Free screen privates after CloseScreen()

The screen privates might still be needed during CloseScreen(), for
example to destroy exa pixmaps.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 dix/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dix/main.c b/dix/main.c
index f98643aa5..e17ff0c7a 100644
--- a/dix/main.c
+++ b/dix/main.c
@@ -310,8 +310,8 @@ dix_main(int argc, char *argv[], char *envp[])
         for (i = screenInfo.numGPUScreens - 1; i >= 0; i--) {
             ScreenPtr pScreen = screenInfo.gpuscreens[i];
             FreeScratchPixmapsForScreen(pScreen);
-            dixFreeScreenSpecificPrivates(pScreen);
             (*pScreen->CloseScreen) (pScreen);
+            dixFreeScreenSpecificPrivates(pScreen);
             dixFreePrivates(pScreen->devPrivates, PRIVATE_SCREEN);
             free(pScreen);
             screenInfo.numGPUScreens = i;
@@ -321,8 +321,8 @@ dix_main(int argc, char *argv[], char *envp[])
             FreeScratchPixmapsForScreen(screenInfo.screens[i]);
             FreeGCperDepth(i);
             FreeDefaultStipple(i);
-            dixFreeScreenSpecificPrivates(screenInfo.screens[i]);
             (*screenInfo.screens[i]->CloseScreen) (screenInfo.screens[i]);
+            dixFreeScreenSpecificPrivates(screenInfo.screens[i]);
             dixFreePrivates(screenInfo.screens[i]->devPrivates, PRIVATE_SCREEN);
             free(screenInfo.screens[i]);
             screenInfo.numScreens = i;
-- 
2.11.0

